perceval/backends/_abstract_backends.py

```diff
@@ -70,6 +70,28 @@ def samples(self, count: int):
 
 
 class AProbAmpliBackend(ABackend):
+
+    def __init__(self):
+        super().__init__()
+        self._cache_iterator = dict()
+
+    def _get_iterator(self, input_state: BasicState, mask=None):
+        n_photons = input_state.n
+
+        if n_photons not in self._cache_iterator.keys():
+            self._cache_iterator[n_photons] = list(allstate_iterator(input_state, mask))
+
+        return self._cache_iterator[n_photons]
+
+    def _reset(self):
```

Comment:
Ok but just make it public and with a more specific name, like `clear_iterator_cache()`.

You don't **have to** use SLOS method names

Reply:
ok